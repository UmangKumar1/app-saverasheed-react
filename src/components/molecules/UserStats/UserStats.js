import React from 'react'
// import PropTypes from 'prop-types'
import styled from 'react-emotion'

// const { any } = PropTypes
const UserStatsRoot = styled('div')({
  padding: 24,
})

const Title = styled('div')({
  fontWeight: 600,
})

const UserStats = () => (
  <UserStatsRoot>
    <Title>Locations Visited:</Title>
    <Title>Quests Completed:</Title>
  </UserStatsRoot>
)

UserStats.propTypes = {}

export default UserStats
