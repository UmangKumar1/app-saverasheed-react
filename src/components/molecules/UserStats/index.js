export { default, default as UserStats } from './UserStats'
