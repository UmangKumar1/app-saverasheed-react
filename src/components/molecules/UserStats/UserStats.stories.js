import React from 'react'
import { storiesOf } from '@storybook/react'
import UserStats from '.'

storiesOf('UserStats', module).add('default', () => (
  <UserStats>UserStats</UserStats>
))
