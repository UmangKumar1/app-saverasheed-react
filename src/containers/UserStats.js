import { connect } from 'react-redux'
import { getUser } from '../modules/auth/selectors'

import { subscribe } from '../util/reselect'
import UserStats from '../components/molecules/UserStats'

const mapStateToProps = subscribe({
  getUser,
})

export default connect()(UserStats)
