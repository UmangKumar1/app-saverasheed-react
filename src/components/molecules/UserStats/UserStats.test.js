import React from 'react'
import { shallow } from 'enzyme'
import UserStats from '.'

const wrap = (props = {}) => shallow(<UserStats {...props} />)

describe('UserStats', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({ children: 'UserStats' })
    expect(wrapper).toMatchSnapshot()
  })
})
