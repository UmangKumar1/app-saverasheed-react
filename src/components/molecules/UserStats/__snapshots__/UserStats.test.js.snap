// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStats renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserStats>
    UserStats
  </UserStats>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(div)>
          Locations Visited:
        </Styled(div)>,
        <Styled(div)>
          Quests Completed:
        </Styled(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Locations Visited:",
        },
        "ref": null,
        "rendered": "Locations Visited:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Quests Completed:",
        },
        "ref": null,
        "rendered": "Quests Completed:",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(div)>
            Locations Visited:
          </Styled(div)>,
          <Styled(div)>
            Quests Completed:
          </Styled(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Locations Visited:",
          },
          "ref": null,
          "rendered": "Locations Visited:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Quests Completed:",
          },
          "ref": null,
          "rendered": "Quests Completed:",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
